address-space-controller/src/main/java/io/enmasse/controller/PodDistruptionBudgetController.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.controller;
+
+import io.enmasse.address.model.AddressSpace;
+import io.enmasse.admin.model.v1.InfraConfig;
+import io.enmasse.admin.model.v1.StandardInfraConfig;
+import io.enmasse.config.AnnotationKeys;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.policy.PodDisruptionBudget;
+import io.fabric8.kubernetes.api.model.policy.PodDisruptionBudgetBuilder;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PodDistruptionBudgetController implements Controller {
+    private static final Logger log = LoggerFactory.getLogger(PodDistruptionBudgetController.class);
+
+    private final NamespacedKubernetesClient client;
+    private final String namespace;
+
+    public PodDistruptionBudgetController(NamespacedKubernetesClient client, String namespace) {
+        this.client = client;
+        this.namespace = namespace;
+    }
+
+    public AddressSpace reconcileActive(AddressSpace addressSpace) throws Exception {
+        InfraConfig infraConfig = InfraConfigs.parseCurrentInfraConfig(addressSpace);
+
+        if (infraConfig instanceof StandardInfraConfig) {
+            RouterSet routerSet = RouterSet.create(namespace, addressSpace, client);
+            reconcileRouterPodDisruptionBudget(addressSpace, routerSet, (StandardInfraConfig) infraConfig);
+            reconcileBrokerPodDisruptionBudget(addressSpace, (StandardInfraConfig) infraConfig);
+        }
+        return addressSpace;
+    }
+
+    private static boolean needsUpdate(IntOrString existing, Integer updated) {
+        return updated != null && (existing == null || !updated.equals(existing.getIntVal()));
+    }
+
+
+    private void reconcileRouterPodDisruptionBudget(AddressSpace addressSpace, RouterSet routerSet, StandardInfraConfig infraConfig) {
+        if (infraConfig.getSpec() != null && infraConfig.getSpec().getRouter() != null && (infraConfig.getSpec().getRouter().getMinAvailable() != null || infraConfig.getSpec().getRouter().getMaxUnavailable() != null)) {
+            String name = String.format("enmasse.%s.%s.qdrouterd", addressSpace.getMetadata().getNamespace(), addressSpace.getMetadata().getName());
+            try {
+                boolean changed = false;
+                PodDisruptionBudget podDisruptionBudget = client.inNamespace(namespace).policy().podDisruptionBudget().withName(name).get();
+                if (podDisruptionBudget == null) {
+                    podDisruptionBudget = new PodDisruptionBudgetBuilder()
+                            .editOrNewMetadata()
+                            .withName(name)
+                            .addToLabels("app", "enmasse")
```

Comment:
I might be missing something here, it doesn't look like there is any selectors specific to the router pods for a specific address space, like there is for the router PDB below?

Reply:
It gets set on line 76

Reply:
Ah sorry, missed that. Thank you